Pixel UI version 1

achalsaharan opened the pull request:
This the first version of my component library, Pixel UI. Looking for code reviews and feedback.

https://pixelui.netlify.app

achalsaharan committed:
Initial Commit

Rashita123 commented on index.html:
```diff
@@ -0,0 +1,58 @@
+    <script src="assets/rainbow-custom.min.js"></script>
+    <link href="./themes/github.css" rel="stylesheet" type="text/css">
+    <title>Pixel UI | Home</title>
+</head>
```
A suggestion:
Add Favicon image, in case you forgot.

Rashita123 commented on index.html:
```diff
@@ -0,0 +1,58 @@
+            <a target="_blank" href="https://linkedin.com/in/achalsaharan"><i class="fab fa-linkedin-in "></i>Twitter</a>
+            <a target="_blank" href="https://twitter.com/achalsaharan_"><i class="fab fa-twitter "></i>LinkedIn</a>    
+        </div>
+    </div>
```
When I'm selecting the component to show from the sidebar, in the mobile view, it is taking me to that component in the docs, but it not removing the sidebar from view.
User will have to click on X to close that.
A suggestion: Make it close when someone has selected the component which he wants to see.

varenaggarwal replied:
Same issue i am facing

Rashita123 commented on index.html:
```diff
@@ -0,0 +1,58 @@
+        <div class="landing-page-image"></div>
+        <div class="landing-page-content">
+            <h1>Pixel UI</h1>
+            <p class="body-cp-lg">Rapidly Build Modern Websites.</p>
```
In the mobile view, we'll have to scroll to left or right, try fixing the width to viewport width, so that we don't have to scroll, i.e. overflow-x should not be there.

Rashita123 commented on styles/avatar.css:
```diff
@@ -0,0 +1,41 @@
+    width: 5rem;
+    border-radius: 50%;
+}
+
```
We can have 2 classes : avatar, which should apply to all the avatars, and then avatar-sm and avatar-lg, which can be applied to any avatar for specific size.
This is just a suggestion, because this is what we usually see.

Here, we have only 1 line of code, which is common to both the sizes, but might lead to repetition if we have more.
Also, avatar-sm and avatar-lg makes it much more easier for the user to understand (that this class is giving just the size property).

Rashita123 commented on styles/styles.css:
```diff
@@ -0,0 +1,249 @@
+    flex-direction: column;
+    color: var(--gray);
+    font-weight: 500;
+    overflow: scroll;
```
The sidebar is showing scrollbars even if we don't need it. So, set overflow to auto. It should solve the issue.

> On using auto, The scrollbars will only show up if there is content that actually breaks out of the element.

> [Read this](https://css-tricks.com/almanac/properties/o/overflow/)

Suggestion: Also set the sidebar to middle of the total height, will look better.

Rashita123 commented on styles/styles.css:
```diff
@@ -0,0 +1,249 @@
+    background: rgb(19, 19, 19, 0.2);
+    width: 100%;
+    height: 100%;
+    z-index: 999;
```
After the session by CSS Wizard, you might want to change this 😆

Set z-index variables, and at max you would require upto 4-5.

Rashita123 commented on styles/styles.css:
```diff
@@ -0,0 +1,249 @@
+:root {
+    --primary: #00bfa5;
+    --primary-dark: #008e76;
+    --primary-light: #5df2d6;
```
We usually set variables for colors, fonts, padding, and margins,
So Ig it is better to have -color, -size, etc after every CSS variable.

E.g.: set --grey-color is a better variable name in my opinion. 

like, variable name --text is not explaining anything about what it is for, if it is text size or text color.

Rashita123 commented on styles/avatar.css:
```diff
@@ -0,0 +1,41 @@
+    left: 1.8rem;
+}
+
+.badge-lg {
```
Same here:
Using classes - btn, btn-sm and btn-lg will make it easier, and you'll have to write lesser code.
A lot of code is being repeated here.

varenaggarwal commented on index.html:
```diff
@@ -0,0 +1,58 @@
+
+    <div class="header-container">
+        <div class="header">
+            <span style="margin-right: 1rem;" id="open-sidebar"><i class="fas fa-bars fa-lg"></i></span>
```
Avoid inline style can move them to css file

varenaggarwal commented on styles/avatar.css:
```diff
@@ -0,0 +1,41 @@
+    height: 1.5rem;
+    width: 1.5rem;
+    background-color: var(--primary-dark);
+    color: #fff;
```
This colour could also be used as variable

varenaggarwal commented on styles/cards.css:
```diff
@@ -0,0 +1,40 @@
+    background-color: var(--primary-dark);
+    color: #fff;
+    position: absolute;
+    z-index: 999;
```
can be a variable and may not need such high number for z-index

varenaggarwal commented on styles/modal.css:
```diff
@@ -0,0 +1,54 @@
+    right: 0;
+    bottom: 0;
+    left: 0;
+    z-index: 999;
```
using the same z-index as card-badge it will create issue. This should me maximum throughout app

varenaggarwal commented on styles/rating.css:
```diff
@@ -0,0 +1,16 @@
+}
+
+.rating-green {
+    background-color: #10b981;
```
can be taken out as a variable

varenaggarwal commented on styles/styles.css:
```diff
@@ -0,0 +1,249 @@
+    margin-bottom: 0;
+}
+
+.btn-float-action {
```
can be part of btn css file

varenaggarwal reviewed (commented):
Great work the home page looks awesome.

achalsaharan committed:
refactor - refactor based on code reviews

Make variables for z-index

Change CSS variable names for better readability

Make sidebar close after clicking the links

achalsaharan committed:
fix - changes based on PR reviews

sakshijainn commented:
Hi 
For Reference:
![image](https://user-images.githubusercontent.com/35770166/113827987-86153e00-97a1-11eb-87cc-83a93db8c0e0.png)
When you click on any component, it is not taking the center div instead the half content for it goes upward : Like for loader it should look like thus when I click Loader 

Also in responsive menu add cursor:pointer for cross button
